docs/src/modeler/time_series.md

```diff
@@ -0,0 +1,283 @@
+```julia
+    resolution = Dates.Hour(1)
+    data = Dict(
+        DateTime("2020-01-01T00:00:00") => ones(24),
+        DateTime("2020-01-01T01:00:00") => ones(24),
+    )
+    forecast = Deterministic("max_active_power", data, resolution)
+```
+- Load from CSV file. For Deterministic forecasts, each row represents one
+lookahead window. The first column must be the initial time and the rest must
+be the forecast values. The CSV file must have no header in the first row.
+
+```julia
+    resolution = Dates.Hour(1)
+    forecast = Deterministic("max_active_power", csv_filename, component, resolution)
+```
+
+## Scaling factors
+Time series data can store actual component values (for instance MW) or scaling
+factors (0 - 1 values). By default PowerSystems treats the values in the time
+series data as physical units. In order to specify them as scaling factors, you
+must pass the accessor function that provides the multiplier value. This value
+must be passed into the forecast when you create it.
+
+Example:
+
+```julia
+    resolution = Dates.Hour(1)
+    data = Dict(
+        DateTime("2020-01-01T00:00:00") => ones(24),
+        DateTime("2020-01-01T01:00:00") => ones(24),
+    )
+    forecast = Deterministic(
+	"max_active_power",
+	data,
+	resolution,
+	scaling_factor_multiplier = get_max_active_power,
+    )
+```
+
+In this example, the forecasted component is a generator. Whenever the user
+retrieves the forecast data PowerSystems will call
+`get_max_active_power(component)` and multiply the result with the forecast
+values (scaling factors). For instance it the maximum active power returns the
+value 50.0 and the scaling factor at some time point is 0.65, the forecast
+value will correspond to 32.5.
+
+## Adding time series to the System
+Adding time series data to a system requires a component that is already
+attached to the system. Extending the example above:
+
+```julia
+    add_time_series!(sys, component, forecast)
+```
+
+In order to optimize the storage of time series data, time series can be shared
+across devices to avoid duplication. If the same forecast applies to multiple
+components then can call `add_time_series!`, passing the collection of
+components that share the time series data.
+
+```julia
+    add_time_series!(sys, components, forecast)
+```
+
+This function stores a single copy of the data. Each component will store a
+reference to that data.
+
+## Removing time series data
+Time series instances can be removed from a system like this:
+
+```julia
+    remove_time_series!(Deterministic, sys, "max_active_power")
+```
+
+**Note**: If you are storing time series data in an HDF5 file, this does
+not actually free up file space (HDF5 behavior). If you want to remove all or
+most time series instances then consider using `clear_time_series!`. It
+will delete the HDF5 file and create a new one. PowerSystems has plans to
+automate this type of workflow.
+
+```julia
+    clear_time_series!(sys)
+```
+
+## Data exploration
```

Comment:
Jose moved this section above the other retrieval sections.  I think I disagree with that move.  I think we should reinforce `get_time_series_array` and `ForecastCache` as the methods to get data.  

1. They account for scaling factors.
2. They will not load all data into memory.

Reply:
I agree with Dan's assessment that we should reinforce  `get_time_series_array` and `ForecastCache`. The reason I moved it is that the first thing people usually want to do after loading data is looking at it.

Reply:
I'm moving this back with a few minor rewording. I think that the original order doesn't prohibit inspecting it... if they really want to see the  data as it's stored, they'll find the next section.